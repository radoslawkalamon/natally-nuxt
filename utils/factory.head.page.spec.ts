import { factoryHeadPage } from '@/utils/factory.head.page'
import { FactoryHeadSchemaWebpageType } from '@/utils/factory.head.schema.webpage'

describe('Utils / Factory / Head / Page', () => {
  test('shall match snapshot | * | WebPage', () => {
    const dto = {
      createdAt: new Date('2021-08-07T16:08:02+00:00'),
      description: 'Rzeczywistość uwypuklona. Staram się spojrzeć z zupełnie innej perspektywy na rzeczy, zjawiska, a nawet słowa.',
      imageOpenGraph: '/images/common.opengraph.png',
      title: 'Poezja 3.14',
      path: '/poezja-314/',
      updatedAt: new Date('2016-12-28T21:23:23+00:00')
    }
    const webpageType = FactoryHeadSchemaWebpageType.WebPage
    expect(factoryHeadPage(dto, webpageType)).toMatchSnapshot()
  })

  test('shall match snapshot | * | CollectionPage', () => {
    const dto = {
      createdAt: new Date('2021-08-07T16:08:02+00:00'),
      description: 'Posłuchaj audiobooków na podstawie moich opowiadań, na swoich ulubionych platformach podcasterskich.',
      imageOpenGraph: '/images/common.opengraph.png',
      title: 'Strona główna',
      path: '/',
      updatedAt: new Date('2016-12-28T21:23:23+00:00')
    }
    const webpageType = FactoryHeadSchemaWebpageType.CollectionPage
    expect(factoryHeadPage(dto, webpageType)).toMatchSnapshot()
  })
})
