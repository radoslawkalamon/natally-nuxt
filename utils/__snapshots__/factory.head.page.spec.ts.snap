// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils / Factory / Head / Page shall match snapshot | * | CollectionPage 1`] = `
{
  "link": [
    {
      "hid": "link:canonical",
      "href": "https://169cm.pl/",
      "rel": "canonical",
    },
  ],
  "meta": [
    {
      "content": "Posłuchaj audiobooków na podstawie moich opowiadań, na swoich ulubionych platformach podcasterskich.",
      "hid": "meta:description",
      "name": "description",
    },
    {
      "content": "Strona główna • 169cm.pl",
      "hid": "meta:og:title",
      "property": "og:title",
    },
    {
      "content": "Posłuchaj audiobooków na podstawie moich opowiadań, na swoich ulubionych platformach podcasterskich.",
      "hid": "meta:og:description",
      "property": "og:description",
    },
    {
      "content": "https://169cm.pl/",
      "hid": "meta:og:url",
      "property": "og:url",
    },
  ],
  "script": [
    {
      "json": {
        "@context": "http://schema.org",
        "@graph": [
          {
            "@id": "https://169cm.pl/#organization",
            "@type": "Organization",
            "image": {
              "@id": "https://169cm.pl/#logo",
            },
            "logo": {
              "@id": "https://169cm.pl/#logo",
              "@type": "ImageObject",
              "caption": "169cm.pl",
              "contentUrl": "https://169cm.pl/images/common.logo.png",
              "height": 224,
              "inLanguage": "pl-PL",
              "url": "https://169cm.pl/images/common.logo.png",
              "width": 448,
            },
            "name": "169cm.pl",
            "sameAs": [
              "https://facebook.com/169cmpl",
              "https://soundcloud.com/169cmpl",
            ],
            "url": "https://169cm.pl/",
          },
          {
            "@id": "https://169cm.pl/#website",
            "@type": "WebSite",
            "description": "Liczy się każda historia!",
            "inLanguage": "pl-PL",
            "name": "169cm.pl",
            "publisher": {
              "@id": "https://169cm.pl/#organization",
            },
            "url": "https://169cm.pl/",
          },
          {
            "@id": "https://169cm.pl/#primaryimage",
            "@type": "ImageObject",
            "contentUrl": "https://169cm.pl/images/common.opengraph.png",
            "height": 630,
            "hid": "schema:ImageObject",
            "inLanguage": "pl-PL",
            "url": "https://169cm.pl/images/common.opengraph.png",
            "width": 1200,
          },
          {
            "@id": "https://169cm.pl/#webpage",
            "@type": "CollectionPage",
            "about": {
              "@id": "https://169cm.pl/#organization",
            },
            "breadcrumb": {
              "@id": "https://169cm.pl/#breadcrumb",
            },
            "dateModified": "2016-12-28T21:23:23.000Z",
            "datePublished": "2021-08-07T16:08:02.000Z",
            "inLanguage": "pl-PL",
            "isPartOf": {
              "@id": "https://169cm.pl/#website",
            },
            "name": "Strona główna • 169cm.pl",
            "potentialAction": [
              {
                "@type": "ReadAction",
                "target": [
                  "https://169cm.pl/",
                ],
              },
            ],
            "primaryImageOfPage": {
              "@id": "https://169cm.pl/#primaryimage",
            },
            "url": "https://169cm.pl/",
          },
          {
            "@id": "https://169cm.pl/#breadcrumb",
            "@type": "BreadcrumbList",
            "itemListElement": [
              {
                "@type": "ListItem",
                "item": "https://169cm.pl/",
                "name": "169cm.pl",
                "position": 1,
              },
            ],
          },
        ],
      },
      "type": "application/ld+json",
    },
  ],
  "title": "Strona główna • 169cm.pl",
}
`;

exports[`Utils / Factory / Head / Page shall match snapshot | * | WebPage 1`] = `
{
  "link": [
    {
      "hid": "link:canonical",
      "href": "https://169cm.pl/poezja-314/",
      "rel": "canonical",
    },
  ],
  "meta": [
    {
      "content": "Rzeczywistość uwypuklona. Staram się spojrzeć z zupełnie innej perspektywy na rzeczy, zjawiska, a nawet słowa.",
      "hid": "meta:description",
      "name": "description",
    },
    {
      "content": "Poezja 3.14 • 169cm.pl",
      "hid": "meta:og:title",
      "property": "og:title",
    },
    {
      "content": "Rzeczywistość uwypuklona. Staram się spojrzeć z zupełnie innej perspektywy na rzeczy, zjawiska, a nawet słowa.",
      "hid": "meta:og:description",
      "property": "og:description",
    },
    {
      "content": "https://169cm.pl/poezja-314/",
      "hid": "meta:og:url",
      "property": "og:url",
    },
  ],
  "script": [
    {
      "json": {
        "@context": "http://schema.org",
        "@graph": [
          {
            "@id": "https://169cm.pl/#organization",
            "@type": "Organization",
            "image": {
              "@id": "https://169cm.pl/#logo",
            },
            "logo": {
              "@id": "https://169cm.pl/#logo",
              "@type": "ImageObject",
              "caption": "169cm.pl",
              "contentUrl": "https://169cm.pl/images/common.logo.png",
              "height": 224,
              "inLanguage": "pl-PL",
              "url": "https://169cm.pl/images/common.logo.png",
              "width": 448,
            },
            "name": "169cm.pl",
            "sameAs": [
              "https://facebook.com/169cmpl",
              "https://soundcloud.com/169cmpl",
            ],
            "url": "https://169cm.pl/",
          },
          {
            "@id": "https://169cm.pl/#website",
            "@type": "WebSite",
            "description": "Liczy się każda historia!",
            "inLanguage": "pl-PL",
            "name": "169cm.pl",
            "publisher": {
              "@id": "https://169cm.pl/#organization",
            },
            "url": "https://169cm.pl/",
          },
          {
            "@id": "https://169cm.pl/poezja-314/#primaryimage",
            "@type": "ImageObject",
            "contentUrl": "https://169cm.pl/images/common.opengraph.png",
            "height": 630,
            "hid": "schema:ImageObject",
            "inLanguage": "pl-PL",
            "url": "https://169cm.pl/images/common.opengraph.png",
            "width": 1200,
          },
          {
            "@id": "https://169cm.pl/poezja-314/#webpage",
            "@type": "WebPage",
            "about": {
              "@id": "https://169cm.pl/#organization",
            },
            "breadcrumb": {
              "@id": "https://169cm.pl/poezja-314/#breadcrumb",
            },
            "dateModified": "2016-12-28T21:23:23.000Z",
            "datePublished": "2021-08-07T16:08:02.000Z",
            "inLanguage": "pl-PL",
            "isPartOf": {
              "@id": "https://169cm.pl/#website",
            },
            "name": "Poezja 3.14 • 169cm.pl",
            "potentialAction": [
              {
                "@type": "ReadAction",
                "target": [
                  "https://169cm.pl/poezja-314/",
                ],
              },
            ],
            "primaryImageOfPage": {
              "@id": "https://169cm.pl/poezja-314/#primaryimage",
            },
            "url": "https://169cm.pl/",
          },
          {
            "@id": "https://169cm.pl/poezja-314/#breadcrumb",
            "@type": "BreadcrumbList",
            "itemListElement": [
              {
                "@type": "ListItem",
                "item": "https://169cm.pl/",
                "name": "169cm.pl",
                "position": 1,
              },
              {
                "@type": "ListItem",
                "item": "https://169cm.pl/poezja-314/",
                "name": "Poezja 3.14",
                "position": 2,
              },
            ],
          },
        ],
      },
      "type": "application/ld+json",
    },
  ],
  "title": "Poezja 3.14 • 169cm.pl",
}
`;
